Makefile

```diff
@@ -36,10 +37,21 @@ endif
 
 LDFLAGS = -s -w -X ${PKG}/version.Version=${VERSION} -X ${PKG}/version.Revision=$(REVISION) -X ${PKG}/version.BuildTimestamp=$(BUILD_TIMESTAMP)
 
+CARGO ?= $(shell which cargo)
+OPTIMIZER_SERVER = tools/optimizer-server
+OPTIMIZER_SERVER_TOML = ${OPTIMIZER_SERVER}/Cargo.toml
+OPTIMIZER_SERVER_BIN = ${OPTIMIZER_SERVER}/target/release/optimizer-server
+
 .PHONY: build
 build:
 	GOOS=${GOOS} GOARCH=${GOARCH} ${PROXY} go build -ldflags "$(LDFLAGS)" -v -o bin/containerd-nydus-grpc ./cmd/containerd-nydus-grpc
 
+.PHONY: build-optimizer
+build-optimizer:
+	${CARGO} fmt --manifest-path ${OPTIMIZER_SERVER_TOML} -- --check
+	${CARGO} build --release --manifest-path ${OPTIMIZER_SERVER_TOML} && cp ${OPTIMIZER_SERVER_BIN} ./bin
```

Comment:
Could you please also add `cargo clippy` to lint the tool, which will give some wise suggestions.

Reply:
Good suggestion. I have added this.